### ISSUE TITLE
Improper Output Device Selection

### ISSUE THREAD

**Issue by xelivous** [tesselode/kira#1]
I tried using the Simple Sound Playback example in the readme with a .wav and .ogg file, but it panics on an internal array access. I then tried using the latest git commit (c2877ebae2034498d94070234fa2db8fd6a2458a) and tweaked the example based on the current changes, but the behaviour was the same.

The panic message: `thread '<unnamed>' panicked at 'index out of bounds: the len is 1 but the index is 1', /home/xelivous/.cargo/git/checkouts/kira-a39b987b43e480e3/c2877eb/kira/src/manager/mod.rs:217:21`

https://github.com/tesselode/kira/blob/c2877ebae2034498d94070234fa2db8fd6a2458a/kira/src/manager/mod.rs#L217

I'm running on Manjaro; I looked at basic issues pertaining to linux/arch on cpal's repo but none of them seem to apply, and i could run cpal's examples just fine on my system.

My best guess from comparing the examples code and the code in kira is that they use `device.default_output_config()?` while kira uses `device.supported_output_configs()?.next()`. The difference being that kira pulls in whatever is listed first, even if it's a very incapable interface that isn't the best suited interface. Debugging it a little, it seems that the interface returned is a single channel interface with a sample rate of `384000`...

If I change the following config selection chain:
https://github.com/tesselode/kira/blob/c2877ebae2034498d94070234fa2db8fd6a2458a/kira/src/manager/mod.rs#L202-L207

Into something like:
```rust
let config = device.default_output_config().ok().ok_or(AudioError::NoSupportedAudioConfig)?.config();
```

Then everything works without problems, although there might be a better way to send out the proper error other than the really dank `ok().ok_or()` combo. And i'm not sure if you did it with `supported_output_configs` for some particular reason that would be broken for other people when using `default_output_config`.

---

The alternative solution is properly handling when an interface only has a single output frame/channel and mapping it to something like mono but idk if running something at a really meme sample rate has value.

**Comment by tesselode** [comment]
Yeah, you're having the same issue another person I'm talking to is having. I didn't have any special reason for using `supported_output_configs`, that's just what I saw in the cpal docs. I can get a fix merged in soon.

**Comment by tesselode** [comment]
I went ahead and released v0.1.2, so this should be fixed now. Let me know if you have any more issues.
